### ISSUE TITLE
Core/Creature: Snake Trap

### ISSUE THREAD

**Issue by Killyana** [TrinityCore/TrinityCore#21847]
**Description:** When Snake Trap is activated by a player or creature that is not in combat with the hunt, snakes will get the unit that triggered the trap as target without attacking it.

**Current behavior:** Snakes from Snake Trap doesn't attack the target if it's not in combat with the hunt

**Expected behavior:** Snakes from Snake Trap must always attack the unit that trigger the trap

**Steps to reproduce the problem:**

1. connect a hunt and a second char from the opposite faction
2. Summon a snake trap .cast 34600
3. trigger the trap by the second char, notice snakes follow you but doesn't attack, If you attack the hunt when this snakes are following you they will attack you back.

**Branch(es):** 3.3.5

TrinityCore rev. 743980a6255a 2018-04-15 17:08:48 +0200 (3.3.5 branch) (Win32, Release, Static)
Using SSL version: OpenSSL 1.0.1e 11 Feb 2013 (library: OpenSSL 1.0.1e 11 Feb 2013)
Using Boost version: 1.59.0
Using MySQL version: 5.5.39
Using CMake version: 3.10.1
Compiled on: Windows 10.0.16299
Automatic database updates are disabled for all databases!
Worldserver listening connections on port 8085
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
MMAPs status: Enabled
maps directory located in ./maps. Total size: 252191207 bytes
vmaps directory located in ./vmaps. Total size: 588247501 bytes
mmaps directory located in ./mmaps. Total size: 2229000480 bytes
Using frFR DBC Locale as default. All available DBC locales: frFR 
Using World DB: TDB 335.64
--->

**Comment by Nitair** [comment]
Seems like they have aggro but no command to attack...? Mh..

**Comment by Jildor** [comment]
Can be a generic problem caused by @Keader commit: https://github.com/TrinityCore/TrinityCore/commit/a891eb364e941a5146fba4af16e1ecf65cdd9239 ?

**Comment by Keader** [comment]
Maybe.
Is Treeston Code btw xD

**Comment by Jildor** [comment]
Oh, yes  sorry.
@Treeston guilty! :smile:

**Comment by Jildor** [comment]
@Treeston can you take a look here?
Maybe we should revert a891eb3 and find a better solution?

**Comment by Treeston** [comment]
Possibly a hack exposed by a891eb3, but definitely not that commit's fault.

I'll take a look.

**Comment by Treeston** [comment]
Yeah, Snake Trap pets are a colossal script. No idea how many hacks are in there, but it's probably a lot.

**Comment by Treeston** [comment]
Reverting 5df7ffa, it is a hack.

Underlying cause is a spell system bug that's causing `Snake Trap Effect` (45145) from the trap GO to not hit the player targeting the trap, which would put them in combat and cause snakes to behave as intended.

Re-opening and assigning to @ariel-
